### PR TITLE
Entity Linking 

### PR DESCRIPTION
**Pull request by vadam5**
This PR adds a new model, dataset, loss, training example code and notebook demo for entity linking. The loss and model are an implementation of the "Self-Alignment Pretraining for Biomedical Entity Representations" paper.

### CODE REVIEW

**Comment by okuchaiev** [comment]
Please ping me offline - let's put the tiny dataset on AWS rather than GitHub repo.
Regarding DCO, when you commit, please add "-s" flag to do signoff on your commits. To sign already committed commits, checkout DCO's bot output https://github.com/NVIDIA/NeMo/pull/2042/checks?check_run_id=2309052982
